### PR TITLE
Add Haproxy Dataplane API 

### PR DESCRIPTION
**Pull request by kulikovav**
### What's this PR about
This PR provides a new package.

Description:
Data Plane API is a sidecar process that runs next to HAProxy and provides API endpoints for managing HAProxy. 
It requires HAProxy version 1.9.0 or higher.

Sources were grabbed from a fedora package.
This spec is not compatible with CentOS 6.

### TODO's:

- [x] Run [`perfecto`](https://github.com/essentialkaos/perfecto) check on your spec file
- [x] Write [`bibop`](https://github.com/essentialkaos/bibop) tests
- [x] Enable "Allow edits from maintainers" for this PR
- [x] Update the messages below

**Did you try to build the package with this spec?:** Yes
**Is this ready for review?:**  Yes

### CODE REVIEW

File: specs/dataplaneapi/dataplaneapi.spec [hunk 1]
```diff
@@ -0,0 +1,144 @@
...
+fi
+
+%preun
+if [[ $1 -eq 0 ]]; then
+  %{__sysctl} --no-reload disable %{name}.service &>/dev/null || :
+  %{__sysctl} stop %{name}.service &>/dev/null || :
+fi
+
+%postun
+  %{__sysctl} daemon-reload &>/dev/null || :
```
Comment (andyone) [inline specs/dataplaneapi/dataplaneapi.spec]: Useless spaces at the beginning of the line.

File: specs/dataplaneapi/dataplaneapi.spec [hunk 2]
```diff
@@ -0,0 +1,144 @@
...
+URL:                https://github.com/haproxytech/dataplaneapi
+
+Source0:            https://github.com/haproxytech/%{name}/archive/v%{version}.tar.gz
+Source1:            %{name}.service
+Source2:            %{name}.logrotate
+Source3:            %{name}.sysconfig
+
+BuildRoot:          %{_tmppath}/%{name}-%{version}-%{release}-root-%(%{__id_u} -n)
+
+BuildRequires:      golang >= 1.12
```
Comment (andyone) [inline specs/dataplaneapi/dataplaneapi.spec]: It is better to use the latest version of go for the build. The latest available version in the release repository is 1.13.7.

File: tests/dataplaneapi.recipe [hunk 3]
```diff
@@ -0,0 +1,33 @@
...
+  app dataplaneapi
+
+  service-present {service_name}
+
+  exist {sysconfig}
+  mode {sysconfig} 644
+  exist {logrotate_config}
+  mode {logrotate_config} 644
+
+command "systemctl start {service_name}" "Start dataplaneapi daemon"
```
Comment (andyone) [inline tests/dataplaneapi.recipe]: If version specific tools are used for the test (_like systemd in this case, which available on CentOS/RHEL >= 7_), you should add the version to file suffix. The proper name for the file will be `dataplaneapi-c7.recipe`

File: specs/dataplaneapi/dataplaneapi.spec [hunk 4]
```diff
@@ -0,0 +1,144 @@
+################################################################################
+
+%global crc_check pushd ../SOURCES ; sha512sum -c %{SOURCE100} ; popd
```
Comment (andyone) [inline specs/dataplaneapi/dataplaneapi.spec:3]: Macro for [automatic CRC generation and check](https://github.com/essentialkaos/rpmbuilder/wiki/Automatic-SHA-512-checksum-generation) defined but there is no `Source100` or macro usage.

**Review by andyone — CHANGES_REQUESTED** [review]

**Comment by kulikovav** [comment]
Such a horrible PR. @andyone, thank you for review. I made the changes.

**Review by andyone — APPROVED** [review]

**Comment by andyone** [comment]
Looks good, merged.

**Comment by andyone** [comment]
The package was built and added to `kaos-testing` repository.

**Pull request merged**